### PR TITLE
[docs] Add description for all pages under SDK in Reference

### PR DESCRIPTION
**Pull request by amandeepmittal**
# Why

Closes ENG-8607

# How

By adding the `description` field in metadata for each doc in SDK reference. Changes backported from unversioned to all existing SDKs. Also, added description for deprecated Firebase libraries in SDK 47 and 46.

# Test Plan

Changes can be tested by running docs locally and visiting a doc in Reference > Expo SDK. For example: http://localhost:3002/versions/latest/sdk/barometer/

**Preview**

CleanShot 2023-06-19 at 19 37 54@2x

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: docs/pages/versions/unversioned/sdk/apple-authentication.mdx [hunk 1]
```diff
@@ -1,5 +1,6 @@
 ---
 title: AppleAuthentication
+description: A library provides Sign in with Apple capability for iOS 13 and higher.
```
Comment (tsapeta) [inline docs/pages/versions/unversioned/sdk/apple-authentication.mdx:3]:
```suggestion
description: A library that provides Sign in with Apple capability for iOS 13 and higher.
```

File: docs/pages/versions/unversioned/sdk/auth-session.mdx [hunk 2]
```diff
@@ -1,5 +1,6 @@
 ---
 title: AuthSession
+description: A universal library provides an API to handle browser-based authentication.
```
Comment (tsapeta) [inline docs/pages/versions/unversioned/sdk/auth-session.mdx:3]:
```suggestion
description: A universal library that provides an API to handle browser-based authentication.
```

File: docs/pages/versions/unversioned/sdk/clipboard.mdx [hunk 3]
```diff
@@ -1,13 +1,14 @@
 ---
 title: Clipboard
+description: A universal library that allows  getting and setting Clipboard content.
```
Comment (tsapeta) [inline docs/pages/versions/unversioned/sdk/clipboard.mdx:3]:
```suggestion
description: A universal library that allows getting and setting Clipboard content.
```

**Review by tsapeta — COMMENTED** [review]
Except missing `that` words, it looks good. However, I would rather be in favor of updating these descriptions in the README first and then copy them to the docs. If we start from duplicating these not very good descriptions, you would have to update them in both places 🤔

File: docs/pages/versions/unversioned/sdk/tracking-transparency.mdx [hunk 4]
```diff
@@ -1,5 +1,6 @@
 ---
 title: TrackingTransparency
+description: A library for requesting permission to track the users on devices using iOS 14.5 and higher.
```
Comment (tsapeta) [inline docs/pages/versions/unversioned/sdk/tracking-transparency.mdx:3]:
```suggestion
description: A library for requesting permission to track the users on devices using iOS 14 and higher.
```

**Review by tsapeta — APPROVED** [review]

File: docs/pages/versions/unversioned/sdk/async-storage.mdx [hunk 5]
```diff
@@ -1,5 +1,6 @@
 ---
 title: AsyncStorage
+description: A library provides an asynchronous, unencrypted, persistent, key-value storage API.
```
Comment (amandeepmittal) [inline docs/pages/versions/unversioned/sdk/async-storage.mdx:3]:
```suggestion
description: A library that provides an asynchronous, unencrypted, persistent, key-value storage API.
```

File: docs/pages/versions/v46.0.0/sdk/apple-authentication.mdx [hunk 6]
```diff
@@ -1,5 +1,6 @@
 ---
 title: AppleAuthentication
+description: A library provides Sign in with Apple capability for iOS 13 and higher.
```
Comment (amandeepmittal) [inline docs/pages/versions/v46.0.0/sdk/apple-authentication.mdx:3]:
```suggestion
description: A library that provides Sign in with Apple capability for iOS 13 and higher.
```

File: docs/pages/versions/v47.0.0/sdk/apple-authentication.mdx [hunk 7]
```diff
@@ -1,5 +1,6 @@
 ---
 title: AppleAuthentication
+description: A library provides Sign in with Apple capability for iOS 13 and higher.
```
Comment (amandeepmittal) [inline docs/pages/versions/v47.0.0/sdk/apple-authentication.mdx:3]:
```suggestion
description: A library that provides Sign in with Apple capability for iOS 13 and higher.
```

File: docs/pages/versions/v48.0.0/sdk/apple-authentication.mdx [hunk 8]
```diff
@@ -1,5 +1,6 @@
 ---
 title: AppleAuthentication
+description: A library provides Sign in with Apple capability for iOS 13 and higher.
```
Comment (amandeepmittal) [inline docs/pages/versions/v48.0.0/sdk/apple-authentication.mdx:3]:
```suggestion
description: A library that provides Sign in with Apple capability for iOS 13 and higher.
```

File: docs/pages/versions/v46.0.0/sdk/async-storage.mdx [hunk 9]
```diff
@@ -1,5 +1,6 @@
 ---
 title: AsyncStorage
+description: A library provides an asynchronous, unencrypted, persistent, key-value storage API.
```
Comment (amandeepmittal) [inline docs/pages/versions/v46.0.0/sdk/async-storage.mdx:3]:
```suggestion
description: A library that provides an asynchronous, unencrypted, persistent, key-value storage API.
```

File: docs/pages/versions/v47.0.0/sdk/async-storage.mdx [hunk 10]
```diff
@@ -1,5 +1,6 @@
 ---
 title: AsyncStorage
+description: A library provides an asynchronous, unencrypted, persistent, key-value storage API.
```
Comment (amandeepmittal) [inline docs/pages/versions/v47.0.0/sdk/async-storage.mdx:3]:
```suggestion
description: A library that provides an asynchronous, unencrypted, persistent, key-value storage API.
```

File: docs/pages/versions/v48.0.0/sdk/async-storage.mdx [hunk 11]
```diff
@@ -1,5 +1,6 @@
 ---
 title: AsyncStorage
+description: A library provides an asynchronous, unencrypted, persistent, key-value storage API.
```
Comment (amandeepmittal) [inline docs/pages/versions/v48.0.0/sdk/async-storage.mdx:3]:
```suggestion
description: A library that provides an asynchronous, unencrypted, persistent, key-value storage API.
```

File: docs/pages/versions/v46.0.0/sdk/clipboard.mdx [hunk 12]
```diff
@@ -1,13 +1,14 @@
 ---
 title: Clipboard
+description: A universal library that allows  getting and setting Clipboard content.
```
Comment (amandeepmittal) [inline docs/pages/versions/v46.0.0/sdk/clipboard.mdx:3]:
```suggestion
description: A universal library that allows getting and setting Clipboard content.
```

File: docs/pages/versions/v47.0.0/sdk/clipboard.mdx [hunk 13]
```diff
@@ -1,13 +1,14 @@
 ---
 title: Clipboard
+description: A universal library that allows  getting and setting Clipboard content.
```
Comment (amandeepmittal) [inline docs/pages/versions/v47.0.0/sdk/clipboard.mdx:3]:
```suggestion
description: A universal library that allows getting and setting Clipboard content.
```

File: docs/pages/versions/v48.0.0/sdk/clipboard.mdx [hunk 14]
```diff
@@ -1,13 +1,14 @@
 ---
 title: Clipboard
+description: A universal library that allows  getting and setting Clipboard content.
```
Comment (amandeepmittal) [inline docs/pages/versions/v48.0.0/sdk/clipboard.mdx:3]:
```suggestion
description: A universal library that allows getting and setting Clipboard content.
```

File: docs/pages/versions/v46.0.0/sdk/tracking-transparency.mdx [hunk 15]
```diff
@@ -1,5 +1,6 @@
 ---
 title: TrackingTransparency
+description: A library for requesting permission to track the users on devices using iOS 14.5 and higher.
```
Comment (amandeepmittal) [inline docs/pages/versions/v46.0.0/sdk/tracking-transparency.mdx:3]:
```suggestion
description: A library for requesting permission to track the users on devices using iOS 14 and higher.
```

File: docs/pages/versions/v47.0.0/sdk/tracking-transparency.mdx [hunk 16]
```diff
@@ -1,5 +1,6 @@
 ---
 title: TrackingTransparency
+description: A library for requesting permission to track the users on devices using iOS 14.5 and higher.
```
Comment (amandeepmittal) [inline docs/pages/versions/v47.0.0/sdk/tracking-transparency.mdx:3]:
```suggestion
description: A library for requesting permission to track the users on devices using iOS 14 and higher.
```

File: docs/pages/versions/v48.0.0/sdk/tracking-transparency.mdx [hunk 17]
```diff
@@ -1,5 +1,6 @@
 ---
 title: TrackingTransparency
+description: A library for requesting permission to track the users on devices using iOS 14.5 and higher.
```
Comment (amandeepmittal) [inline docs/pages/versions/v48.0.0/sdk/tracking-transparency.mdx:3]:
```suggestion
description: A library for requesting permission to track the users on devices using iOS 14 and higher.
```

**Pull request merged**